sbin/upload-artifacts

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+
+PROGNAME="${BASH_SOURCE[0]}"
+HERE="$(cd "$(dirname "$PROGNAME")" &>/dev/null && pwd)"
+ROOT=$(cd $HERE/.. && pwd)
+READIES=$ROOT/deps/readies
+. $READIES/shibumi/defs
+
+#----------------------------------------------------------------------------------------------
+
+if [[ $1 == --help || $1 == help || $HELP == 1 ]]; then
+	cat <<-END
+		Upload packages to S3
+
+		upload-artifacts [--help|help] artifacts...
+
+		Argument variables:
+		SNAPSHOT=1    Upload snapshots (default)
+		OSNICK=nick   Operate on packages for given OSNICK (default: host)
+
+		RELEASE=1     Upload release artifacts
+
+		FORCE=1       Allow uploading outside of CI
+		NOP=1         No operation
+		HELP=1        Show help
+
+	END
+	exit 0
+fi
+
+#----------------------------------------------------------------------------------------------
+
+ARCH=$($READIES/bin/platform --arch)
+[[ $ARCH == x64 ]] && ARCH="x86_64"
+
+OS=$($READIES/bin/platform --os)
+[[ $OS == linux ]] && OS="Linux"
+
+[[ -z $OSNICK ]] && OSNICK=$($READIES/bin/platform --osnick)
+[[ $OSNICK == trusty ]] && OSNICK=ubuntu14.04
+[[ $OSNICK == xenial ]] && OSNICK=ubuntu16.04
+[[ $OSNICK == bionic ]] && OSNICK=ubuntu18.04
+[[ $OSNICK == focal ]] && OSNICK=ubuntu20.04
+[[ $OSNICK == centos7 ]] && OSNICK=rhel7
+[[ $OSNICK == centos8 ]] && OSNICK=rhel8
+
+PLATFORM="$OS-$OSNICK-$ARCH"
+
+#----------------------------------------------------------------------------------------------
+
+OP=""
+[[ $NOP == 1 ]] && OP=echo
+
+if [[ $STAGING == 1 ]]; then
+	S3_URL=s3://redismodules/lab/staging
+else
+	S3_URL=s3://redismodules
+fi
+
+if [[ $FORCE != 1 ]]; then
+	if [[ -z $CIRCLECI ]]; then
+		eprint "Cannot upload outside of CircleCI. Override with FORCE=1."
+		exit 1
+	fi
+
+	if [[ -z $AWS_ACCESS_KEY_ID || -z $AWS_SECRET_ACCESS_KEY ]]; then
+		eprint "No credentials for S3 upload."
+		exit 1
+	fi
+fi
+
+cd $ROOT/bin
+
+if [[ $SNAPSHOT == 1 ]]; then
+	SUBDIR=/snapshots
+else
+	SUBDIR=
+fi
+
+cd artifacts${SUBDIR}
+[[ $VERBOSE == 1 ]] && du -ah --apparent-size *
+
+#----------------------------------------------------------------------------------------------
+
+s3_upload_file() {
+	local file="$1"
+	local s3_dir="$2"
+	[[ $s3_dir != */ ]] && s3_dir="${s3_dir}/"
+	
+	$OP python2 -m awscli s3 cp $file $s3_dir --acl public-read --no-progress
```

Comment:
You probably can call awscli directly without python2 here.

Reply:
Yes, I know. I think I reacted to a plain `awscli`. I'll revert this after this is more stable.